doc/sphinx/language/extensions/implicit-arguments.rst

```diff
@@ -284,61 +282,137 @@ Declaring Implicit Arguments
 
-   The :n:`@arguments_modifier` values have various effects:
+   .. todo need a definition for "maximally inserted"  Maybe here??
+   .. todo is the RHS in the next 3 overkill?
 
-   * :n:`clear implicits` - clears implicit arguments
-   * :n:`default implicits` - automatically determine the implicit arguments of the object.
+   * :n:`( ... ) {? % @scope }` -
+     :n:`(@name__1 @name__2 ...)%@scope` is shorthand for :n:`@name__1%@scope @name__2%@scope ...`.
+   * :n:`[ ... ] {? % @scope }` - declares the enclosed names as implicit.
+     :n:`[@name__1 @name__2 ... ]%@scope` is equivalent to :n:`[@name__1]%@scope [@name__2]%@scope ...`.
+
+   .. todo "Arguments Nat.add [n]%nat m%nat." is accepted.  Is it useful?
+
+   * :n:`%{ ... %} {? % @scope }` - declares the enclosed names as maximally inserted.
+     :n:`%{@name__1 @name__2 ... %}%@scope` is equivalent to :n:`%{@name__1%}%@scope %{@name__2%}%@scope ...`.
+
+   * `!` - the constant will be unfolded only if all the arguments marked with `!`
+     evaulate to constructors.  See :tacn:`simpl`.
+   * :n:`@name {? % @scope }` - a *formal parameter* of the function :n:`smart_qualid` (i.e.
+     the parameter name used in the function definition).  Unless `rename` is specified,
+     the list of :n:`@name`\s must be a prefix of the formal parameters.  `_` can be used
+     to skip over a formal parameter.
+     :token:`scope` can be either a scope name or its delimiting key.
+
+   .. todo: how does the prefix work with parameters that were implicit in the function declaration??
+
+   * `clear implicits` - makes all implicit arguments into explicit arguments
+   * `default implicits` - automatically determine the implicit arguments of the object.
      See :ref:`auto_decl_implicit_args`.
-   * :n:`rename` - rename implicit arguments for the object
-   * :n:`assert` - assert that the object has the expected number of arguments with the
+   * `rename` - rename implicit arguments for the object
+   * `assert` - assert that the object has the expected number of arguments with the
      expected names.  See the example here: :ref:`renaming_implicit_arguments`.
 
+     .. todo: "Arguments Nat.add {n}%bool x%nat : assert." also changes n to {n}%nat.
+        feature or bug?  describe?
+
+   * `clear scopes` - clears argument scopes of :n:`@smart_qualid`
+   * `extra scopes` - defines extra argument scopes, to be used in case of coercion to ``Funclass``
+     (see the :ref:`implicitcoercions` chapter) or with a computed type.
+   * `simpl nomatch` - prevents performing a simplification step for :n:`@smart_qualid`
+     that would expose a match construct in the head position.  See :tacn:`simpl`.
+   * `simpl never` - prevents performing a simplification step for :n:`@smart_qualid`.
+     See :tacn:`simpl`.
+
+     .. todo: simpl nomatch/simpl never apply only to tactics simpl and cbn, or to others
+        as well (based on how simpl nomatch is described in the tactics chapter)?
```

Comment:
I guess the examples under `simpl` should be moved to its own spot (maybe in the description of `Eval`?) and referenced from those 3 tactics (which perhaps should mostly just point to `Eval`).  Indeed, the tactics could just reference `Eval`.  Your thoughts?

And I'll mention that it doesn't apply to the others. (You think that's a feature or a bug?)

Reply:
It is likely that the description of `red_expr` will be split in several parts, as some parts reference reduction functions provided by the kernel, while other use some Ltac syntax. See also the on-going discussion at #11679.

> And I'll mention that it doesn't apply to the others. (You think that's a feature or a bug?)

It makes sense that some reduction strategies cannot be controlled, but I don't know whether *more* reduction strategies should be controlled.

Reply:
I tweaked the wording at `simpl` in `tactics.rst` to include `hnf`.  I expect this will be better explained when we get better descriptions of the reductions.